src/Uno.UI/UI/Xaml/Controls/Popup/PopupBase.cs

```diff
@@ -110,18 +110,25 @@ protected internal override void OnTemplatedParentChanged(DependencyPropertyChan
 
 		private void UpdateDataContext()
 		{
-			_childHasOwnDataContext = false;
 			if (Child is IDependencyObjectStoreProvider provider)
 			{
-				var dataContextProperty = provider.Store.ReadLocalValue(provider.Store.DataContextProperty);
-				if (dataContextProperty != null && dataContextProperty != DependencyProperty.UnsetValue)
+				if (DataContext == null)
 				{
-					// Child already has locally set DataContext, we shouldn't overwrite it.
-					_childHasOwnDataContext = true;
+					provider.Store.ClearValue(provider.Store.DataContextProperty);
```

Comment:
When I saw this line, I expected it'd break something, and indeed:

https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=26832&view=ms.vss-test-web.build-test-results-tab&runId=1357172&resultId=100300&paneView=debug

> Assert.AreEqual failed. Expected:<Starfish>. Actual:<(null)>.
>at Uno.UI.Tests.ContentDialogTests.Given_ContentDialog.When_Has_DataContext() in \src\Uno.UI.Tests\Windows_UI_XAML_Controls\ContentDialogTests\Given_ContentDialog.cs:line 36

This fix won't work because you're overwriting the DataContext that might've been explicitly set on the child, which is exactly what the code here is guarding against.

Reply:
Updated to a less intrusive fix (for now)